lib/src/checks.dart

```diff
@@ -1,4 +1,51 @@
+import 'package:nyxx/nyxx.dart';
 import 'package:nyxx_commands/nyxx_commands.dart';
 import 'package:running_on_dart/src/settings.dart';
 
 final administratorCheck = UserCheck.anyId(adminIds, 'Administrator check');
+
+final connectedToAVoiceChannelCheck = Check((IContext context) async {
+  final selfMember = await context.guild!.selfMember.getOrDownload();
+
+  if (selfMember.voiceState == null || selfMember.voiceState!.channel == null) {
+    await context.respond(MessageBuilder.content('I have to be in a voice channel to use this command'));
+    return false;
+  }
+  return true;
+}, "musicConnectedToVC");
+
+final notConnectedToAVoiceChannelCheck = Check((IContext context) async {
+  final selfMember = await context.guild!.selfMember.getOrDownload();
+
+  if (selfMember.voiceState == null || selfMember.voiceState!.channel == null) {
+    return true;
+  }
+  await context.respond(MessageBuilder.content("I'm already connected to a voice channel"));
+  return false;
+}, "musicNotConnectedToVC");
+
+final sameVoiceChannelOrDisconnectedCheck = Check((IContext context) async {
+  // If this is an interaction, acknowledge it just in case the check
+  // takes too long to run.
+  if (context is InteractionChatContext) {
+    await context.acknowledge();
+  }
+
+  final selfMemberVoiceState = (await context.guild!.selfMember.getOrDownload()).voiceState;
+  final memberVoiceState = context.member!.voiceState;
+
+  if (memberVoiceState == null || memberVoiceState.channel == null) {
+    await context.respond(MessageBuilder.content('You need to be connected to a voice channel to use this command'));
+    return false;
+  }
+
+  if (selfMemberVoiceState == null || selfMemberVoiceState.channel == null) {
+    return true;
+  }
+
+  if (selfMemberVoiceState.channel!.id != memberVoiceState.channel!.id) {
+    await context.respond(MessageBuilder.content("I'm already being used on other voice channel"));
```

Comment:
Same here, this should be in a separate check as the above comment, which can then be combined with `Check.all()` since they are always used as a pair.

Reply:
what should be in a separate check?

Reply:
Each check should only validate one thing about the state - this check does two, as can be seen by the presence of two distinct responses.

Reply:
Already solved